.github/workflows/deploy_pages.yml

```diff
@@ -0,0 +1,215 @@
+name: Deploy to GitHub Pages
+on:
+  workflow_run:
+    workflows: [CI]
+    types:
+      - completed
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+  deployments: write
+  statuses: write
+  checks: write
+
+jobs:
+  upload:
+    if: ${{ github.repository == 'primer/react' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create github-pages directory
+        run: mkdir github-pages
+      - name: Download storybook
```

Comment:
I am not sure if I understand why we need to download storybook here 🤔  We won't deploy it to github-pages and we separately deploy it as well. I am sure I miss something here, would be great to clarify this 😊

Reply:
@broccolinisoup the way I understand it, for GH Pages deploy previews we will need to download all the artifacts that should be in the preview and then place them under this `github-pages` directory which will act as the root for serving files.

My thought process was that the output of `docs` would be the default set of files served when visiting a deploy preview URL, and then we would add folders for each sub-resource like `/storybook` for Storybook, `/vrt` for Visual Regression Testing, `/axe` for Axe, etc.

Hope this makes sense, let me know if you have any other questions or if I'm misunderstanding anything here 👀